pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+
+            * **context** (:func:`py:tuple` of server_context, namespace):
+              A context object identifying the open enumeration session,
+              including its current enumeration state, and the namespace. This
+              object must be supplied with the next pull or close operation for
+              this enumeration session.
+
+              The tuple items are:
+
+              * server_context (:term:`string`):
+                Enumeration context string returned by the server if
+                the session is not exhausted, or `None` otherwise. This string
+                is opaque for the client.
+              * namespace (:term:`string`):
+                Name of the CIM namespace that was used for this operation.
+
+              NOTE: This inner tuple hides the need for a CIM namespace
+              on subsequent operations in the enumeration session. CIM
+              operations always require target namespace, but it never
+              makes sense to specify a different one in subsequent
+              operations on the same enumeration session.
+
+
+        Raises:
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+            CIMError: CIM_ERR_INVALID_PARAMETER
+
+        """
+
+        self._test_pull_operations_disabled()
+        self._validate_open_params(FilterQueryLanguage, FilterQuery,
+                                   OperationTimeout, ContinueOnError)
+
+        result = self.EnumerateInstances(
+            namespace, ClassName,
+            DeepInheritance=DeepInheritance,
+            IncludeQualifiers=None,              # not used with this request
+            IncludeClassOrigin=IncludeClassOrigin,
+            PropertyList=PropertyList)
+
+        return self._open_response(namespace, result,
+                                   'PullInstancesWithPath',
+                                   OperationTimeout,
+                                   MaxObjectCount,
+                                   ContinueOnError)
+
+    ####################################################################
+    #
+    #   OpenReferenceInstancePaths server request
+    #
+    ####################################################################
+
+    def OpenReferenceInstancePaths(self, namespace, InstanceName,
+                                   ResultClass=None, Role=None,
+                                   FilterQueryLanguage=None, FilterQuery=None,
+                                   OperationTimeout=None, ContinueOnError=None,
+                                   MaxObjectCount=None):
+        # pylint: disable=invalid-name
+        """
+        Implements WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.OpenReferenceInstancePaths`
+        with data from the instance CIM repository.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
```

Comment:
Copy & paste error from CreateInstance. Should use enumeration verbiage.

Reply:
DONE